pkg/resourceinterpreter/interpreter.go

```diff
@@ -49,31 +51,39 @@ type ResourceInterpreter interface {
 }
 
 // NewResourceInterpreter builds a new ResourceInterpreter object.
-func NewResourceInterpreter(kubeconfig string, informer genericmanager.SingleClusterInformerManager) ResourceInterpreter {
+func NewResourceInterpreter(
+	informer genericmanager.SingleClusterInformerManager,
+	restMapper meta.RESTMapper,
+	scaleClient scale.ScalesGetter,
+) ResourceInterpreter {
 	return &customResourceInterpreterImpl{
-		kubeconfig: kubeconfig,
-		informer:   informer,
+		informer:    informer,
+		restMapper:  restMapper,
+		scaleClient: scaleClient,
 	}
 }
 
 type customResourceInterpreterImpl struct {
-	kubeconfig string
-	informer   genericmanager.SingleClusterInformerManager
+	informer    genericmanager.SingleClusterInformerManager
+	restMapper  meta.RESTMapper
+	scaleClient scale.ScalesGetter
 
-	customizedInterpreter *customizedinterpreter.CustomizedInterpreter
-	defaultInterpreter    *defaultinterpreter.DefaultInterpreter
+	customizedInterpreter  *customizedinterpreter.CustomizedInterpreter
+	defaultInterpreter     *defaultinterpreter.DefaultInterpreter
+	subresourceInterpreter *subresourceinterpreter.SubResourceInterpreter
```

Comment:
Another approach would be to put the `subresourceInterpreter` into `defaultInterpreter`.

Reply:
Yes, that's okay. Do I need to adjust it?